generators/profilegen/src/profile_generator.rb

```diff
@@ -0,0 +1,225 @@
+# encoding: utf-8
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+
+require 'erb'
+require 'json'
+require 'ms_rest'
+require 'fileutils'
+require 'ms_rest_azure'
+require_relative 'profile_templates'
+
+#
+# Class to generate the profile
+#
+
+class ProfileGenerator
+  # Accessors to be used in the generation of profile client
+  attr_accessor :file_names, :profile_name, :class_names, :individual_gem_profile
+  # Accessors to be used in the generation of module
+  attr_accessor :module_require, :module_name, :class_name, :operation_types, :management_client, :model_types, :versions_clients_mapper
+  # Accessors to be used in the generation of profile spec file
+  attr_accessor :profile_version, :profile_ruby_version, :spec_includes, :module_name
+
+  def initialize(profile, dir_metadata)
+    @profile_name = profile['name']
+    @resource_provider_types = profile['resourceTypes']
+    @profile_ruby_version = profile['ruby_version']
```

Comment:
seems like unused `@profile_ruby_version` for now...do you have any forward looking case for this?

Reply:
removed it. Excellent catch :-)